python_modules/dagster/dagster/_core/definitions/materialize.py

```diff
@@ -59,7 +95,11 @@ def materialize(
 
     JOB_NAME = "__ephemeral_asset_job__"
 
-    defs = Definitions(jobs=[define_asset_job(name=JOB_NAME)], assets=assets, resources=resources)
+    defs = Definitions(
+        jobs=[define_asset_job(name=JOB_NAME, selection=selection)],
```

Comment:
Do we need the `define_asset_job` given `Definitions.get_implicit_global_asset_job_def`?

Reply:
We could find a way to work around it, but it makes things easier, because otherwise we would need to:
- Find the right implicit asset job (there can be multiple if different assets have different partitions)
- Coerce the asset selection selection

Open to changing it, but seems orthogonal to what's going on in this PR.

Reply:
true, I just noticed bc the line was green but now I realize it was already there